extensions/alice-ai/src/commands.tsx

```diff
@@ -0,0 +1,23 @@
+import { Detail, LaunchProps } from "@raycast/api";
+import "./lib/OpenAI";
+import { useActionsAreReady } from "./store/actions";
+import CommandExecute from "./views/CommandExecute";
+import CommandList from "./views/CommandList";
+
+interface Arguments {
+  id?: string;
+}
+
+export default function ActionsCommand(props: LaunchProps<{ arguments: Arguments }>) {
+  const ready = useActionsAreReady();
+
+  if (!ready) {
+    return <Detail />;
+  }
+
+  if (props.arguments.id !== undefined) {
+    return <CommandExecute id={props.arguments.id} />;
+  }
```

Comment:
We can maybe remove all this and just send ready={ready} to CommandList which as `isLoading` as property instead

Reply:
The extension's state is persisted to LocalStorage, and every time the command is called, data is rehydrated from LocalStorage to the state. Unfortunately, this process is asynchronous, and I need to wait for the rehydrate process to finish before I can use the data (my useActionsAreReady hook).

In this command, I have two responsibilities:
1. Displaying the list of actions - which is default.
2. Immediately proceeding to execute an action by calling the command created using "Create QuickLink".

In both cases, the store must already be loaded to handle the request correctly. If I moved the loading to CommandList, I would skip CommandExecute, which might not work due to the extension state not being loaded.